pandas/core/groupby/ops.py

```diff
@@ -1214,7 +1214,11 @@ def names(self) -> list[Hashable]:
     @property
     def groupings(self) -> list[grouper.Grouping]:
         lev = self.binlabels
-        ping = grouper.Grouping(lev, lev, in_axis=False, level=None)
+        codes = self.group_info[0]
+        labels = lev.take(codes)
+        ping = grouper.Grouping(
+            labels, labels, in_axis=False, level=None, uniques=lev.values
```

Comment:
is there any danger that lev.values makes a copy?  e.g. i think tzaware DatetimeIndex might

Reply:
Thanks - this indeed did not work for tzaware. Changed to ._values and added tests.